### PR TITLE
fixed wrong max handling

### PR DESCRIPTION
**Pull request by jaboll-ai**
#### Small thing here
If you re-read the line you know what went wrong. If the width should increase but flows over the max width, it should be capped at the max-width (maxWidth-currentWidth) instead of (what previously happened) stay at the currentWidth. 

The width expansion request is still valid, it should increase, even for a tiny but, but stop at the cap. It should NOT stop at the point it was where it then will hit the cap.

TL;DR consistency with setting a maxWidth

### CODE REVIEW

**Comment by jaboll-ai** [comment]
Heyo. Thanks for looking over it. Before releasing it tho let me check again today. I think I found another issue 😕

**Comment by jaboll-ai** [comment]
Nope. Works as intended

**Comment by Quorafind** [comment]
ok
